app/src/main/java/de/danoeh/antennapod/fragment/SwipeActions.java

```diff
@@ -0,0 +1,344 @@
+        };
+
+        itemTouchHelper = new ItemTouchHelper(simpleCallback);
+    }
+
+    private static int actionIconFor(String swipeAction) {
+        switch (swipeAction) {
+            case ADD_TO_QUEUE:
+                return R.drawable.ic_playlist;
+            case MARK_PLAYED:
+                return R.drawable.ic_check;
+            default:
+            case MARK_UNPLAYED:
+                return R.drawable.ic_check;
+            case MARK_FAV:
+                return R.drawable.ic_star;
+            case START_DOWNLOAD:
+                return R.drawable.ic_download;
+        }
+    }
+
+    private static int actionColorFor(String swipeAction) {
+        switch (swipeAction) {
+            case ADD_TO_QUEUE:
+                return R.color.swipe_light_green_200;
+            case MARK_PLAYED:
+                return R.color.swipe_light_red_200;
+            default:
+            case MARK_UNPLAYED:
+                return R.color.swipe_light_blue_200;
+            case MARK_FAV:
+                return R.color.swipe_yellow_200;
+            case START_DOWNLOAD:
+                return R.color.swipe_light_blue_200;
+        }
+    }
+
+    private static String[] getPrefs(Context context, String tag, String defaultActions) {
+        SharedPreferences prefs = context.getSharedPreferences(PREF_NAME, Context.MODE_PRIVATE);
+        String prefsString = prefs.getString(PREF_SWIPEACTIONS + tag, defaultActions);
+        if (prefsString.isEmpty()) {
+            //no prefs set
+            return new String[]{};
+        }
+        return prefsString.split(",");
+    }
+
+    private static String[] getPrefs(Context context, String tag) {
+        return getPrefs(context, tag, "");
+    }
+
+    private static String[] getPrefsWithDefaults(Context context, String tag) {
+        String defaultActions;
+        switch (tag) {
+            /*case InboxFragment.TAG:
+                defaultActions = ADD_TO_QUEUE + "," + MARK_UNPLAYED;
+                break;*/
+            default:
+            case EpisodesFragment.TAG:
+                defaultActions = MARK_FAV + "," + START_DOWNLOAD;
+                break;
+        }
+
+        return getPrefs(context, tag, defaultActions);
+    }
+
+    public void resetItemTouchHelper() {
+        //prevent swipe staying if item is staying in the list
+        if (itemTouchHelper != null && recyclerView != null) {
+            itemTouchHelper.attachToRecyclerView(null);
+            itemTouchHelper.attachToRecyclerView(recyclerView);
+        }
+    }
+
+    public void show() {
+        new SwipeActionsDialog().show(() -> {
+            //detach old
+            itemTouchHelper.attachToRecyclerView(null);
+            //create new
+            itemTouchHelper();
+            //attach new
+            attachTo(recyclerView);
+        });
+    }
+
+    private void show(Callback callback) {
+        new SwipeActionsDialog().show(callback);
+    }
+
+    private class SwipeActionsDialog {
```

Comment:
Please move this to its own class. I actually would make this a settings page instead (instead of having it always present in the overflow menu).

Reply:
settings page could be optional, i'd definetly keep the dialog though for the first time swipe, so the user understands that there are options (or just hit ok for the default).